### PR TITLE
Update to GHC 9.10.1

### PR DESCRIPTION
**Pull request by jasagredo**

### CODE REVIEW

File: quickcheck-state-machine.cabal [hunk 1]
```diff
@@ -23,7 +23,7 @@ extra-doc-files:
   README.md
 
 tested-with:
-  GHC ==8.8.4 || ==8.10.7 || ==9.2.8 || ==9.4.7 || ==9.6.3 || ==9.8.1
```
Comment (stevana) [inline quickcheck-state-machine.cabal:26]: `tested-with` should correspond to this list: https://github.com/stevana/quickcheck-state-machine/blob/master/.github/workflows/main.yaml#L23 , please keep exact versions and add 9.10.1 to CI.

File: quickcheck-state-machine.cabal [hunk 2]
```diff
@@ -70,14 +70,15 @@ library no-vendored-treediff
...
-    , graphviz        >=2999.20.0.3 && <2999.21
-    , pretty-show     >=1.6.16      && <1.11
-    , QuickCheck      >=2.12        && <2.15
-    , random          >=1.1         && <1.3
-    , sop-core        >=0.5.0.2     && <0.6
-    , split           >=0.2.3.5     && <0.3
-    , unliftio        >=0.2.7.0     && <0.3
+    , graphviz                     >=2999.20.0.3 && <2999.21
+    , pretty-show                  >=1.6.16      && <1.11
+    , prettyprinter                ^>=1.7.1
```
Comment (stevana) [inline quickcheck-state-machine.cabal:75]: Does this PR also resolve #44?

**Review by stevana — CHANGES_REQUESTED** [review]

File: quickcheck-state-machine.cabal (hunk 1 above)
Comment (jasagredo) [reply quickcheck-state-machine.cabal:26]: Done

File: quickcheck-state-machine.cabal (hunk 2 above)
Comment (jasagredo) [reply quickcheck-state-machine.cabal:75]: It does indeed.

**Comment by jasagredo** [comment]
Sadly, the test-suite is not buildable with GHC 9.10.1. I have added a `if (ghc >= 9.10.1) buildable: False` in the cabal description.

**Comment by stevana** [comment]
If the tests don't pass with 9.10.1, then 9.10.1 isn't supported and should be removed from the changelog, CI and tested-with.

**Comment by jasagredo** [comment]
Its not that they don't pass. The dependencies are not buildable with 9.10.1. But I'm fine with your statement. Will remove it and undo the related changes.

**Pull request closed without merging**